web/controllers/form_columns_controller.ex

```diff
@@ -7,33 +7,38 @@ defmodule Formerer.FormColumnsController do
   alias Formerer.{Submission, Repo}
 
   def edit(conn, %{"forms_id" => id}) do
-    form = get_user_form(current_user(conn), id)
-    submission = first_submission(form)
-
-    conn
-    |> put_layout(false)
-    |> render("edit.html", form: form, submission: submission)
+    case current_user(conn) |> get_user_form(id) do
+      { :ok, form } ->
+        submission = first_submission(form)
+        conn
+        |> put_layout(false)
+        |> render("edit.html", form: form, submission: submission)
+      { :error, error } ->
+        send_resp(conn, 404, error)
+    end
   end
 
   def update(conn, %{"columns" => columns, "forms_id" => id}) do
-    form = get_user_form(current_user(conn), id)
-
-    case update_columns(form, columns) do
-      { :ok, _ } ->
-        json(conn, %{
-            success: true,
-            url: forms_path(conn, :show, form),
-            message: "Columns updated!"
-        })
-      { :error, changeset} ->
-        conn
-        |> put_status(400)
-        |> json(%{
-          success: false,
-          message: changeset.errors
-        })
+    case current_user(conn) |> get_user_form(id) do
+      { :ok, form } ->
+        case update_columns(form, columns) do
```

Comment:
I know this nesting was my doing originally, I think it'd be good to move it to a private function, happy to do it after this is merged if you remind me though